internal/core/window.rs

```diff
@@ -359,6 +372,76 @@ impl Window {
         }
     }
 
+    /// Take the focus_item out of this Window
+    ///
+    /// This sends the FocusOut event!
+    fn take_focus_item(self: Rc<Self>) -> Option<ItemRc> {
```

Comment:
Since this function doesn’t seem to need to consume or clone the Rc, I think it can either take a &self or &Rc<Self>. Then you can save the extra clone call at the call site(s).

Reply:
Good point, going to change this.